sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/aad/AADClientRegistrationRepository.java

```diff
@@ -3,83 +3,402 @@
+        this.delegatedClientRegistrations = delegatedClientRegistrations();
+        this.allClientRegistrations = allClientRegistrations();
+    }
+
+    private AzureClientRegistration azureClientRegistration() {
+        if (!isWebAppClientMode && !isAllInClientMode) {
+            return emptyAzureClientRegistration();
+        }
+
+        AuthorizationClientProperties azureProperties =
+            properties.getAuthorizationClients()
+                      .getOrDefault(AZURE_CLIENT_REGISTRATION_ID, defaultAzureAuthorizationClientProperties());
+        ClientRegistration.Builder builder = toClientRegistrationBuilder(AZURE_CLIENT_REGISTRATION_ID,
+            azureProperties);
+        Set<String> authorizationCodeScopes = authorizationCodeScopes();
+        ClientRegistration client = builder.scope(authorizationCodeScopes).build();
+        Set<String> accessTokenScopes = accessTokenScopes();
+        if (resourceServerCount(accessTokenScopes) == 0 && resourceServerCount((authorizationCodeScopes)) > 1) {
+            // AAD server will return error if:
+            // 1. authorizationCodeScopes have more than one resource server.
+            // 2. accessTokenScopes have no resource server
+            accessTokenScopes.add(properties.getGraphBaseUri() + "User.Read");
+        }
+        return new AzureClientRegistration(client, accessTokenScopes);
+    }
+
+    private Map<String, ClientRegistration> delegatedClientRegistrations() {
+        if (!isWebAppClientMode && !isAllInClientMode) {
+            return Collections.emptyMap();
+        }
+
+        return properties.getAuthorizationClients()
+                         .entrySet()
+                         .stream()
+                         .filter(entry -> isAzureDelegatedClientRegistration(entry.getKey(), entry.getValue()))
+                         .collect(Collectors.toMap(
+                             Map.Entry::getKey,
+                             entry -> toClientRegistration(entry.getKey(), entry.getValue())));
+    }
+
+    private Map<String, ClientRegistration> allClientRegistrations() {
+        Map<String, ClientRegistration> result =
+            properties.getAuthorizationClients()
+                      .entrySet()
+                      .stream()
+                      .filter(entry -> !delegatedClientRegistrations.containsKey(entry.getKey()))
+                      .collect(Collectors.toMap(
+                          Map.Entry::getKey,
+                          entry -> toClientRegistration(entry.getKey(), entry.getValue())));
+        result.putAll(delegatedClientRegistrations);
+        ClientRegistration azureClient = azureClientRegistration.getClient();
+        result.put(AZURE_CLIENT_REGISTRATION_ID, azureClient);
+        return Collections.unmodifiableMap(result);
+    }
+
+    private void handleDefaultAADAuthorizationGrantType(String registrationId,
+                                                        AuthorizationClientProperties properties) {
```

Comment:
How about moving this to `AADAuthenticationProperties#afterPropertiesSet`, it's better to all configuration check(default value setting) in one place.

Reply:
Maybe not suitable, it depends on the application type, so the more business login will be processed in properties class.